### PR TITLE
Adding function to add a user and corresponding test cases. Adding errorcodes in a separate file.

### PR DESCRIPTION
**Pull request by sjindal-arista**
Minimal inital commit, will be adding rest of the user api functions shortly.  For now, only a handful of error codes added which might be useful.

### CODE REVIEW

**Comment by sjindal-arista** [comment]
@cheynearista @shachiagarwal

File: api/user_test.go [hunk 1]
```diff
@@ -201,3 +202,57 @@ func Test_CvpGetUserValid_UnitTest(t *testing.T) {
...
+	client := NewMockClient("", nilUserErr)
+	api := NewCvpRestAPI(client)
+	expectedErr := errors.Errorf("AddUser: %s", nilUserErr.Error())
+
+	if err := api.AddUser(nil); err.Error() != expectedErr.Error() {
+		t.Fatalf("Expected error: [%v] But received: [%v]", expectedErr, err)
+	}
+}
+
+func Test_CvpAddUserAlreadyExists_UnitTest(t *testing.T) {
```
Comment (adarsh-pandey) [inline api/user_test.go:216]: Why not use table driven tests here too? That is easier to maintain and add new tests.

File: api/user.go [hunk 2]
```diff
@@ -109,3 +109,29 @@ func (c CvpRestAPI) GetUser(userID string) (*SingleUser, error) {
...
+// AddUser adds 'user' and returns error if any
+func (c CvpRestAPI) AddUser(user *SingleUser) error {
+	if user == nil {
+		return errors.New("AddUser: can not add nil user")
+	}
+	resp, err := c.client.Post("/user/addUser.do", nil, user)
+	if err != nil {
+		return errors.Errorf("AddUser: %s", err)
+	}
+	var addedUser *SingleUser
```
Comment (adarsh-pandey) [inline api/user.go:122]: nit: user

File: api/user_test.go [hunk 3]
```diff
@@ -32,8 +32,9 @@
 package cvpapi
```
Comment (adarsh-pandey) [inline api/user_test.go:32]: In later commits, we can add assert package here too for better failure handling.

**Review by adarsh-pandey — COMMENTED** [review]
Thanks for aggressively pushing on this, Shivam. Mostly looks good to me. Few minor nits.

File: api/user_test.go (hunk 1 above)
Comment (cheynearista) [reply api/user_test.go:216]: +1 on table test

**Review by cheynearista — APPROVED** [review]
Overall, additions look good, so I'll go ahead and approve.

File: api/user_test.go (hunk 1 above)
Comment (sjindal-arista) [reply api/user_test.go:216]: By table test, you mean put test cases as subtests in 1 single AddUser test, right ?

File: api/user.go (hunk 2 above)
Comment (sjindal-arista) [reply api/user.go:122]: that variable is supposed to have user data that was returned as response to the api call. 
I named it so, as it reflects the user which was added by cvp into the D.B.

File: api/user_test.go (hunk 3 above)
Comment (sjindal-arista) [reply api/user_test.go:32]: Yep, we can add that. Asserts are always useful in testing.

File: api/user_test.go (hunk 1 above)
Comment (sjindal-arista) [reply api/user_test.go:216]: This could be done in later commits. I have raised an issue regarding this.

**Pull request merged**